crates/ironrdp-web/src/session.rs

```diff
@@ -159,27 +158,27 @@ impl SessionBuilder {
         self.clone()
     }
 
-    /// Required
-    pub fn hide_pointer_callback(&self, callback: js_sys::Function) -> SessionBuilder {
-        self.0.borrow_mut().hide_pointer_callback = Some(callback);
-        self.clone()
-    }
-
-    /// Required
-    pub fn hide_pointer_callback_context(&self, context: JsValue) -> SessionBuilder {
-        self.0.borrow_mut().hide_pointer_callback_context = Some(context);
+    /// Required.
+    ///
+    /// #### Callback signature:
+    /// ```typescript
+    /// function callback(cursor_kind: string, hotspot_x: number, hotspot_y: number): void
```

Comment:
Suggestion: for `cursor_kind`, use an enum instead of a string.
Rationale: self documented and type checked (less error prone).

For that, add a new variant "url", and a new parameter "cursor_data" which is something like: `string | undefined`.

Reply:
Great idea